### PR TITLE
Change default ports

### LINKED ISSUE #990: port conflict?
It seems the existing bitcoin ports are being used by default.
hopefully there are plans to change this to make running zcashd alongside bitcoind easier

### PR DESCRIPTION
**Pull request by str4d**
Closes #990.

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -418,7 +418,7 @@ std::string HelpMessage(HelpMessageMode mode)
     strUsage += HelpMessageOpt("-rpcbind=<addr>", _("Bind to given address to listen for JSON-RPC connections. Use [host]:port notation for IPv6. This option can be specified multiple times (default: bind to all interfaces)"));
     strUsage += HelpMessageOpt("-rpcuser=<user>", _("Username for JSON-RPC connections"));
     strUsage += HelpMessageOpt("-rpcpassword=<pw>", _("Password for JSON-RPC connections"));
-    strUsage += HelpMessageOpt("-rpcport=<port>", strprintf(_("Listen for JSON-RPC connections on <port> (default: %u or testnet: %u)"), 8332, 18332));
+    strUsage += HelpMessageOpt("-rpcport=<port>", strprintf(_("Listen for JSON-RPC connections on <port> (default: %u or testnet: %u)"), 8232, 18232));
```
Comment (daira) [inline src/init.cpp]: Where is this changed in the code?

File: src/chainparams.cpp [hunk 2]
```diff
@@ -156,7 +156,7 @@ class CTestNetParams : public CMainParams {
         pchMessageStart[2] = 0xe7;
         pchMessageStart[3] = 0x55;
         vAlertPubKey = ParseHex("04302390343f91cc401d56d68b123028bf52e5fca1939df127f63c6467cdf9c8e2c14b61104cf817d0b780da337893ecc4aaff1309e536162dabbdb45200ca2b0a");
-        nDefaultPort = 18333;
+        nDefaultPort = 18233;
```
Comment (daira) [inline src/chainparams.cpp]: I Googled this port and it's not used by anything important.

File: src/chainparams.cpp [hunk 3]
```diff
@@ -53,7 +53,7 @@ class CMainParams : public CChainParams {
         pchMessageStart[2] = 0x4e;
         pchMessageStart[3] = 0xd8;
         vAlertPubKey = ParseHex("04fc9702847840aaf195de8442ebecedf5b095cdbb9bc716bda9110971b28a49e0ead8564ff0db22209e0374782c093bb899692d524e9d6a6956e7c5ecbcd68284");
-        nDefaultPort = 8333;
+        nDefaultPort = 8233;
```
Comment (daira) [inline src/chainparams.cpp]: I Googled this port and it's used by iTunes, but that's probably not a significant clash.

**Comment by daira** [comment]
NACK because I don't see where the JSON-RPC ports are changed in the code.

**Comment by str4d** [comment]
I missed a few files. Commit updated with the JSON-RPC port changes.

File: src/chainparams.cpp (hunk 3 above)
Comment (str4d) [inline src/chainparams.cpp]: If you look at neighbouring ports, iTunes comes up too. So it's likely a range of ports that is randomly-selected from, so IMHO not significant at all.

**Comment by daira** [comment]
ACK. We will need to mention the port changes in the release notes.

**Comment by ebfull** [comment]
The RPC test framework doesn't need to be updated for this?

**Comment by str4d** [comment]
I saw no references to the default port numbers there, which means either they use the default and will change automatically, or specify a port (like the performance tests do) and wouldn't change anyway.

**Comment by str4d** [comment]
Hmm, searching on GitHub shows a bunch more places with the old port listed, but it's all docs or helper scripts/config files we don't use. I can change all these too though if desired.

**Comment by daira** [comment]
Yes, let's change the docs and scripts.

**Comment by str4d** [comment]
Commit updated to change all docs and scripts in addition to code. Tests are left as-is because they aren't relying on the defaults.

**Comment by bitcartel** [comment]
Ports 8233 and 18233 are unassigned according to IANA, however they are visually similar to the existing bitcoin ports of 8333 and 18333 (and RPC ports 8332 and 18332).  Easy for someone to make a typo.  How about ports 38333 and 48333 where the prefix would clearly distinguish zcash from bitcoin?

**Comment by str4d** [comment]
We decided to stick with ports 8233 and 18233 (and correspondingly 8232 and 18232 for JSON-PRC) for the z5 release.

**Comment by daira** [comment]
Post-hoc ACK

**Pull request merged**